### PR TITLE
Simplify tests for upward derivative

### PR DESCRIPTION
**Pull request by santisoler**
**Description**

Ditch the precomputed sample grid file and add new fixtures for creating
a pair of sample sources, a sample grid and some gravity fields as
DataArrays. Remove `netcdf4` from testing dependencies.

**Relevant issues/PRs**

Related to #299

### CODE REVIEW

**Comment by LL-Geo** [comment]
> @LL-Geo how does it looks now? Does it makes a good boiler plate for the test you're working on in #299 ?

Hey @santisoler , that looks very good for me! I'm thinking to add a little complexity to the test. It might be good to put a pair of sources with positive and negative density contrast and depth.

File: harmonica/tests/test_transformations.py [hunk 1]
```diff
@@ -7,75 +7,135 @@
...
+
+@pytest.fixture(name="sample_sources")
+def fixture_sample_sources():
+    """
+    Define a pair of sample point sources used to build tests
+    """
+    points = [
+        [-50e3, 50e3],
+        [-50e3, 50e3],
+        [-20e3, -20e3],
```
Comment (LL-Geo) [inline harmonica/tests/test_transformations.py:28]: might be good to have one shallow source and one deep source?

File: harmonica/tests/test_transformations.py [hunk 2]
```diff
@@ -7,75 +7,135 @@
...
+    ]
+
+    masses = [3.3e8, 2.9e8]
```
Comment (LL-Geo) [inline harmonica/tests/test_transformations.py:31]: A positive and negative mass might be good as well?

File: harmonica/tests/test_transformations.py (hunk 1 above)
Comment (santisoler) [reply harmonica/tests/test_transformations.py:28]: Sure! Well thought!

File: harmonica/tests/test_transformations.py (hunk 2 above)
Comment (santisoler) [reply harmonica/tests/test_transformations.py:31]: Agreed! You're right that the model should be a little bit more complex.

**Comment by santisoler** [comment]
Thanks @LL-Geo for the review! Making the sample model more complex was a clever idea. I'm merging this!

**Pull request merged**